packages/nextra-theme-docs/src/components/sidebar.tsx

```diff
@@ -322,33 +321,38 @@ export function Sidebar({
       ) : null}
       <aside
         className={cn(
-          'nextra-sidebar-container nextra-scrollbar',
-          'fixed top-16 z-[15] h-[calc(100vh-4rem)] w-full flex-shrink-0 self-start overflow-y-auto md:sticky md:w-64',
+          'nextra-sidebar-container',
+          'fixed inset-y-16 z-[15] w-full flex-shrink-0 overflow-y-auto md:sticky',
+          'motion-reduce:transform-none',
           asPopover ? 'md:hidden' : 'md:block',
-          '[&::-webkit-scrollbar-track]:mt-[var(--nextra-navbar-height)] md:[&::-webkit-scrollbar-track]:mt-5',
-          {
-            open: menu,
-            '[&::-webkit-scrollbar-track]:mb-[var(--nextra-menu-height)]':
-              hasMenu
-          }
+          hasMenu && 'with-menu',
+          menu && 'open',
+          'nextra-scrollbar w-full md:w-64'
         )}
         ref={containerRef}
       >
-        <div
-          className="h-full w-full select-none pl-[calc(env(safe-area-inset-left)-1.5rem)] md:h-auto [-webkit-touch-callout:none]"
-          ref={sidebarRef}
-        >
-          <div className="min-h-[calc(100vh-4rem-61px)] p-4">
-            <div
-              className={cn(
-                'sticky top-0 z-[1] block md:hidden -mt-4 mb-4 bg-white pt-4',
-                'dark:bg-dark shadow-[0_2px_14px_6px_#fff] dark:shadow-[0_2px_14px_6px_#111]'
-              )}
-            >
-              {renderComponent(config.search.component, {
-                directories: flatDirectories
-              })}
-            </div>
+        <div className="h-full pl-[calc(env(safe-area-inset-left)-1.5rem)] md:h-auto [-webkit-touch-callout:none]">
+          <div className="min-h-[calc(100vh-var(--nextra-navbar-height)-var(--nextra-menu-height))] py-4 px-2">
+            {menu ? (
```

Comment:
it's better to avoid rendering unnecessary `Menu` from mobile on desktop and from desktop on mobile